rskj-core/src/main/java/org/ethereum/vm/VM.java

```diff
@@ -767,20 +768,23 @@ protected void doCODESIZE() {
             spendOpCodeGas();
         }
         // EXECUTION PHASE
-        int length;
-        if (op == OpCode.CODESIZE)
-            //TODO(mmarquez): we need to add support to precompiled contracts
-        {
-            length = program.getCode().length; // during initialization it will return the initialization code size
+        DataWord codeLength;
+        if (op == OpCode.CODESIZE) {
+            codeLength = new DataWord(program.getCode().length); // during initialization it will return the initialization code size
         } else {
             DataWord address = program.stackPop();
-            length = program.getCodeAt(address).length;
+            codeLength = new DataWord(program.getCodeAt(address).length);
+            if (program.getBlockchainConfig().isRfs90()) {
+                PrecompiledContracts.PrecompiledContract precompiledContract = PrecompiledContracts.getContractForAddress(config, address);
+                if (precompiledContract != null) {
+                    codeLength = new DataWord(BigIntegers.asUnsignedByteArray(DataWord.MAX_VALUE));
```

Comment:
I thought we agreed on using a small number, e.g., `1`

Reply:
Yes, I'm sorry, I should have put this into the comments section of the PR. While I was implementing this I asked @donequis for his guidance and this matter came up. He told me, and I agreed, that at some point this value could be used for something different than just checking if there is a contract at a given address or not, and a small number here may allow that usage in the context of a precompiled contract while a really big number shouldn't